core/src/main/java/hudson/model/DownloadService.java

```diff
@@ -324,6 +324,21 @@ public String getUrl() {
         }
 
         /**
+         * URL of specific updateSite.
+         */
+        public String getUrl(UpdateSite site) {
```

Comment:
+1. It's location in Download service smells like a hack and needs justification